modules/apps/frontend-js/frontend-js-clay-sample-web/src/main/resources/META-INF/resources/js/App.js

```diff
@@ -12,20 +12,54 @@
  * details.
  */
 
-import ClayAlert from '@clayui/alert';
 import React from 'react';
+import useLiferayStore from 'frontend-js-web/liferay/store/useLiferayStore.es';
 
 import '../css/main.scss';
 
+const UPDATE_NAME = 'UPDATE_NAME';
+const UPDATE_AGE = 'UPDATE_AGE';
+
+const reducer = (state, action) => {
+	const newState = {};
+	if (action.type === UPDATE_NAME) {
+		newState.name = action.value;
+	} else if (action.type === UPDATE_AGE) {
+		newState.age = action.value;
+	}
+
+	return {
+		...state,
+		...newState,
+	};
+};
+
 export default () => {
+	const {state, dispatch} = useLiferayStore('test', reducer, {
+		name: 'Foo',
+		age: 20,
+	});
```

Comment:
Another option for this API would be to just mimic `useState`

```jsx
const [state, setState] = useLiferayStore('test');
```

and then internally we would call "dispatch". This would need some re-working of reducers and likely not rely on those.